frontend/ui/src/pages/Inbox/MessageInput/index.tsx

```diff
@@ -189,54 +196,95 @@ const MessageInput = (props: MessageInputProps & ConnectedProps<typeof connector
     );
   };
 
+  const getLastMessageWithSuggestedReplies = useCallback(() => {
+    const lastMessages = props.messages
+      ?.filter((message: Message) => message.senderType == 'source_contact')
+      .slice(props.messages.length - 5)
+      .reverse();
+    return lastMessages?.find(
+      (message: Message) => message.metadata?.suggestions && Object.keys(message.metadata.suggestions).length > 0
+    );
+  }, [props.messages]);
```

Comment:
So the idea here is that if the last message does not have suggestions we assume that any of the previous 5 message might have suggestions that will also work? :) 

I don't really have an opinion on whether or not that's good. Just want to see if I understand the idea.

Reply:
Yes, it was outlined in the ticket this way.